src/allencell_ml_segmenter/core/file_input_widget.py

```diff
@@ -214,14 +214,18 @@ def _from_directory_slot(self) -> None:
         self._model.set_input_mode(InputMode.FROM_PATH)
 
     def _update_layer_list(self, event: Optional[NapariEvent] = None) -> None:
-        existing_selection = [
-            self._viewer.get_layers_nonthreshold()[i].name
-            for i in self._image_list.get_checked_rows()
-        ]
+        existing_selection: List[str] = []
+        for i in self._image_list.get_checked_rows():
+            item: Optional[QListWidgetItem] = self._image_list.item(i)
+            if item is not None:
+                existing_selection.append(item.text())
+
         self._image_list.clear()
         self._reset_channel_combobox()
         for layer in self._viewer.get_layers():
-            path_of_layer_image: str = layer.source.path
+            path_of_layer_image: Optional[Path] = self._viewer.get_source_path(
+                layer
+            )
             if path_of_layer_image:
                 self._image_list.add_item(
                     layer.name, set_checked=layer.name in existing_selection
```

Comment:
It seems weird that `FileInputWidget` relies on `Viewer` for file paths.  I guess it is because when a file is drag/dropped, the Viewer is the first place where that new state can be found?

I think this widget might be better as a single component with the viewer, since they seemed tightly coupled.

Reply:
yeah